### PR TITLE
run CI on all PR branches

### PR DESCRIPTION
**Pull request by andrewkolos**
This is a follow-up to https://github.com/dart-lang/tools/pull/280#issuecomment-2189941102. In summary, we only run CI checks on PRs against the main branch. However, in cases where an older major version of a package needs to be patched, it doesn't make sense to PR against main. CI won't run for these PRs, but it probably should.

This updates our GitHub CI configs to run checks against PRs against any branch.

---

- [X] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -4,7 +4,8 @@ name: Publish
 
 on:
   pull_request:
-    branches: [ main ]
+    branches:
+      - '**'
```
Comment (andrewkolos) [inline .github/workflows/publish.yaml:8]: I can only guess how/when this is used. I'm not sure if I should have just left this alone.

Comment (devoncarew) [reply .github/workflows/publish.yaml:8]: I would leave the publish workflow as is (there are other checks - within the publishing tool - that also check for `main`).

**Review by devoncarew — COMMENTED** [review]
Thanks for the update! lgtm modulo the change to the publishing workflow file.

**Review by devoncarew — APPROVED** [review]

**Comment by andrewkolos** [comment]
FYI: I've created a release for unified_analytics-v6.1.2 since it looks like that never got made after https://github.com/dart-lang/tools/pull/277 merged. I did this as I didn't want this PR merging to trigger a release from CI.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**